geo/src/algorithm/intersects/mod.rs

```diff
@@ -1,278 +1,114 @@
-use crate::algorithm::contains::Contains;
-use crate::{Line, LineString, Point, Polygon, Rect};
-use num_traits::Float;
-
-/// Checks if the geometry A intersects the geometry B.
-
+use crate::*;
+
+/// Checks if the geometry A intersects the geometry B. More
+/// formally, either boundary or interior of A has non-empty
+/// (set-theoretic) intersection with the boundary or
+/// interior of B. In other words, the [DE-9IM] intersection
```

Comment:
Thanks for documenting this! I hope we can get all the traits to this level of specification.